### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by Yoland172**
test pull request

### CODE REVIEW

File: src/assets/icon/SearchArrow.jsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const SearchArrow = () => {
+    return (
+<svg width="24" height="24">
```
Comment (Anonim147) [inline src/assets/icon/SearchArrow.jsx]: Keep formatting

File: src/assets/icon/SearchArrow.jsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    <path d="M11.293 4.707 17.586 11H4v2h13.586l-6.293 6.293 1.414 1.414L21.414 12l-8.707-8.707-1.414 1.414z"/>
+</svg>)
+}
+
+
```
Comment (Anonim147) [inline src/assets/icon/SearchArrow.jsx]: remove one line

File: src/assets/icon/LoadingIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { IconProp } from "../../types/types"
+
+
+const LoadingIcon = ({width,height}:IconProp) => {
+    return (
+        <svg version="1.1" id="Layer_1"  x="0px" y="0px"
+                            width={width} height={height} viewBox="0 0 24 30" >
```
Comment (Anonim147) [inline src/assets/icon/LoadingIcon.tsx]: keep formatting

File: src/assets/icon/leftArrow.jsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const LeftArrow = ({width, height}) => {
```
Comment (Anonim147) [inline src/assets/icon/leftArrow.jsx]: add type for props
if you use {widht, height} you can create something generic like 
```
interface IconProps {
     width: number;
    height: number;
} 
```
and use this interface for all icons/images

File: src/assets/icon/leftLongArrow.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+const  LeftLongArrow = () => {
+    return (
```
Comment (Anonim147) [inline src/assets/icon/leftLongArrow.tsx]: keep formatting

File: src/components/filmPage/filmInfo/filmInfo.module.scss [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+      width: fit-content;
+      height: auto;
+      border: 2px solid var(--golden);
+      border-radius: 12px;
+      box-sizing: border-box;
+      overflow: hidden;
+      margin-left: 40px;
+
+      .goldContainer {
+        // rename contanier
```
Comment (Anonim147) [inline src/components/filmPage/filmInfo/filmInfo.module.scss:112]: do not forget

File: src/components/homePage/Search/__tests__/Search.test.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    fireEvent.change(input, { target: { value: "test" } });
+
+    //Assert
+    expect(tryToFindFilmsByTyipingMock).toHaveBeenCalledWith("test");
+    expect(tryToFindFilmsByTyipingMock).toBeCalledTimes(1);
+    expect(setSearchTextMock).toBeCalledTimes(1);
+    expect(setSearchTextMock).toHaveBeenCalledWith("test");
+  });
+
+  test("pres button working", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/Search.test.tsx]: typo: press

File: src/components/homePage/Search/__tests__/Search.test.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import Search from "../Search";
+
+describe("Search component", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/Search.test.tsx]: rename to "Search component tests"

File: src/components/homePage/Search/__tests__/Search.test.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  test("renders Search", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/Search.test.tsx]: rename to "Search should render correctly"

File: src/components/homePage/Search/__tests__/Search.test.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+        tryFindFilmBySearch={jest.fn()}
+        tryToFindFilmsByTyiping={jest.fn()}
+        setSearchText={jest.fn()}
+      />
+    );
+    const input = screen.getByPlaceholderText("Search...");
+    expect(input).toBeInTheDocument();
+  });
+
+  test("onchange working", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/Search.test.tsx]: rename to "OnChange should be called with correct params "

File: src/components/homePage/mainHeader/mainHeader.tsx [hunk 11]
```diff
@@ -0,0 +1,66 @@
+import styles from "./mainHeader.module.scss";
+import superman_header from "../../../assets/img/superman_header.jpg";
+import batman_header from "../../../assets/img/batman_header.jpg";
+import batma_header1 from "../../../assets/img/batman_headerCopy.jpg";
+import Slider from "react-slick";
+import "slick-carousel/slick/slick.css";
+import "slick-carousel/slick/slick-theme.css";
+
+function MainHeader() {
+  const slides = [
```
Comment (Anonim147) [inline src/components/homePage/mainHeader/mainHeader.tsx]: Move slides to separated file and pas it through the props

File: src/components/homePage/mainHeader/mainHeader.tsx [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+      name: "Batman I",
+      year: "2003",
+    },
+    {
+      img: batman_header,
+      name: "Batman II",
+      year: "2001",
+    },
+  ];
+  const settings = {
```
Comment (Anonim147) [inline src/components/homePage/mainHeader/mainHeader.tsx]: The same as above for settigns + add type

File: src/components/homePage/mainHeader/mainHeader.tsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+    autoplay: true,
+    arrows: false,
+    pauseOnHover: true,
+    autoplaySpeed: 4000,
+  };
+
+  return (
+    <div className={styles.roundaboutContainer}>
+      <Slider {...settings}>
+        {slides.map((e) => {
```
Comment (Anonim147) [inline src/components/homePage/mainHeader/mainHeader.tsx]: Do not forget about key when rendering multiple elements
<div key={key}>

File: src/assets/icon/SearchArrow.jsx (hunk 1 above)
Comment (Anonim147) [inline src/assets/icon/SearchArrow.jsx]: Do not use magic numbers like 24
Pass it through the props

File: src/assets/icon/leftLongArrow.tsx [hunk 14]
```diff
@@ -0,0 +1,10 @@
...
+        <svg width="68" height="24" viewBox="0 0 68 24"  xmlns="http://www.w3.org/2000/svg">
+<path d="M0.939339 10.9393C0.353554 11.5251 0.353554 12.4749 0.939339 13.0607L10.4853 22.6066C11.0711 23.1924 12.0208 23.1924 12.6066 22.6066C13.1924 22.0208 13.1924 21.0711 12.6066 20.4853L4.12132 12L12.6066 3.51472C13.1924 2.92894 13.1924 1.97919 12.6066 1.3934C12.0208 0.807616 11.0711 0.807616 10.4853 1.3934L0.939339 10.9393ZM68 10.5L2 10.5L2 13.5L68 13.5L68 10.5Z" fill="white"/>
+</svg>
+
```
Comment (Anonim147) [inline src/assets/icon/leftLongArrow.tsx]: Remove this line

File: src/assets/icon/rightArrow.jsx [hunk 15]
```diff
@@ -0,0 +1,11 @@
+
+const RightArrow = ({width, height}) => {
+return (
+<svg width={width} height={height} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 25 25">
+    <path d="m17.5 5.999-.707.707 5.293 5.293H1v1h21.086l-5.294 5.295.707.707L24 12.499l-6.5-6.5z" data-name="Right"/>
+</svg>
+
```
Comment (Anonim147) [inline src/assets/icon/rightArrow.jsx]: Remove this line and keep format

File: src/assets/icon/rightLongArrow.tsx [hunk 16]
```diff
@@ -0,0 +1,11 @@
+const  RightLongArrow = () => {
+    return (
+        <svg width="68" height="24" viewBox="0 0 68 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+<path d="M67.0607 13.0607C67.6464 12.4749 67.6464 11.5251 67.0607 10.9394L57.5147 1.39341C56.9289 0.807621 55.9792 0.807621 55.3934 1.39341C54.8076 1.97919 54.8076 2.92894 55.3934 3.51473L63.8787 12L55.3934 20.4853C54.8076 21.0711 54.8076 22.0208 55.3934 22.6066C55.9792 23.1924 56.9289 23.1924 57.5147 22.6066L67.0607 13.0607ZM-2.62268e-07 13.5L66 13.5L66 10.5L2.62268e-07 10.5L-2.62268e-07 13.5Z" fill="white"/>
+</svg>
+
```
Comment (Anonim147) [inline src/assets/icon/rightLongArrow.tsx]: remove 2 lines and format

File: src/assets/icon/socialMedia/InstagramIcon.tsx [hunk 17]
```diff
@@ -0,0 +1,18 @@
+import { IconProp } from "../../../types/types";
+
```
Comment (Anonim147) [inline src/assets/icon/socialMedia/InstagramIcon.tsx:2]: Sometimes you use 2 lines between imports and code. Sometimes only 1. Please keep one line or two in all places

File: src/components/app/App.test.tsx [hunk 18]
```diff
@@ -1,8 +1,8 @@
-import React from 'react';
-import { render, screen } from '@testing-library/react';
-import App from './App';
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import App from "./App";
 
-test('renders learn react link', () => {
+test("renders learn react link", () => {
```
Comment (Anonim147) [inline src/components/app/App.test.tsx]: Are you really need this test? 
This one related to the tutorial, so you probably need to remove it

File: src/components/filmPage/filmInfo/FilmInfo.tsx [hunk 19]
```diff
@@ -0,0 +1,177 @@
...
+          ) : (
+            <></>
+          )}
+          <div className={styles.ratingList}>
+            <div className={styles.ratingItem}>
+              <h1>IMDB Rating:</h1>
+              <div className={styles.iconWithRating}>
+                <p>
+                  {filmInfo?.imdbRating} / <span>10</span>
+                  {/* <span className={styles.votes}>
```
Comment (Anonim147) [inline src/components/filmPage/filmInfo/FilmInfo.tsx]: Remove comments in while file

File: src/components/filmPage/filmInfo/FilmInfoContainer.tsx [hunk 20]
```diff
@@ -0,0 +1,40 @@
+import { useParams } from "react-router-dom";
+import FilmInfo from "./FilmInfo";
+import { useEffect, useState } from "react";
+import { getFilmInfo } from "../../../api/requests";
+import { FilmItemInfoPage } from "../../../types/types";
+import FilmInfoSkeleton from "./FilmInfoSkeleton";
+
+const FilmInfoContainer = () => {
+  const filmId = useParams().id;
```
Comment (Anonim147) [inline src/components/filmPage/filmInfo/FilmInfoContainer.tsx]: const { id } = useParams();

File: src/components/filmPage/filmInfo/FilmInfoSkeleton.tsx [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+          <Skeleton width="50vw" height="4vh" borderRadius="15px" />
+          <div className={styles.addInfoContainer}>
+            <Skeleton width="15vw" height="4vh" borderRadius="15px" />
+            <Skeleton width="15vw" height="4vh" borderRadius="15px" />
+            <Skeleton width="15vw" height="4vh" borderRadius="15px" />
+            <Skeleton width="15vw" height="4vh" borderRadius="15px" />
+            <Skeleton width="15vw" height="4vh" borderRadius="15px" />
+            <Skeleton width="15vw" height="4vh" borderRadius="15px" />
+          </div>
+          {/*filmInfo?.Writer != "N/A" ?
```
Comment (Anonim147) [inline src/components/filmPage/filmInfo/FilmInfoSkeleton.tsx]: Remove comments please

File: src/components/filmPage/filmInfo/filmInfo.module.scss [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+
+      display: flex;
+      justify-content: center;
+      align-items: center;
+
+      svg rect {
+        fill: var(--golden);
+      }
+    }
+    //styles for skeleton
```
Comment (Anonim147) [inline src/components/filmPage/filmInfo/filmInfo.module.scss]: This one duplicates comment above. Remove both or rename one of them

File: src/components/homePage/Search/search.module.scss [hunk 23]
```diff
@@ -0,0 +1,270 @@
...
+        transition: 300ms;
+      }
+    }
+  }
+}
+
+// Film list container
+
+.mainFilmListContainer {
+  // must be new name of class!!!
```
Comment (Anonim147) [inline src/components/homePage/Search/search.module.scss]: do not forget

File: src/components/homePage/mainHeader/mainHeader.module.scss [hunk 24]
```diff
@@ -0,0 +1,147 @@
...
+.imageContainer {
+  width: 100%;
+  height: 100%;
+  transition: 200ms opacity ease-in-out;
+
+  & > img {
+    width: 100%;
+    height: 100%;
+    object-fit: cover;
+    transition: 200ms opacity ease-in-out; // not-working animation
```
Comment (Anonim147) [inline src/components/homePage/mainHeader/mainHeader.module.scss]: if not working then why you keep it?

File: src/components/homePage/mainHeader/mainHeader.module.scss [hunk 25]
```diff
@@ -0,0 +1,147 @@
...
+  padding: 40px 0px 0px 40px;
+  & > p {
+    color: var(--light-grey);
+    font-size: 5vw;
+    letter-spacing: 4px;
+    font-weight: 550;
+  }
+}
+
+///// not-working animation
```
Comment (Anonim147) [inline src/components/homePage/mainHeader/mainHeader.module.scss]: remove it please if not working

File: src/index.tsx [hunk 26]
```diff
@@ -1,12 +1,13 @@
...
 import ReactDOM from 'react-dom/client';
 import App from './components/app/App';
+import { BrowserRouter } from 'react-router-dom';
 
 const root = ReactDOM.createRoot(
   document.getElementById('root') as HTMLElement
 );
 root.render(
-  <React.StrictMode>
+  <BrowserRouter>
```
Comment (Anonim147) [inline src/index.tsx:10]: Why you removed React.StrictMode?

File: src/components/filmPage/filmInfo/__tests__/filmInfo.test.tsx [hunk 27]
```diff
@@ -0,0 +1,114 @@
...
+    imdbID: "tt1877830",
+    Type: "movie",
+    DVD: "19 Apr 2022",
+    BoxOffice: "$369,345,583",
+    Production: "N/A",
+    Website: "N/A",
+    Response: "True",
+  };
+
+  test("component is render", () => {
```
Comment (Anonim147) [inline src/components/filmPage/filmInfo/__tests__/filmInfo.test.tsx]: Component matches shapshot

File: src/components/homePage/Search/__tests__/FilmList.test.tsx [hunk 28]
```diff
@@ -0,0 +1,176 @@
...
+          />
+        </BrowserRouter>
+      )
+      .toJSON();
+
+    //Assert
+    expect(tree).toMatchSnapshot();
+  });
+
+  test("pagination is render and working", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/FilmList.test.tsx]: Please separate this tests.
One test for render and one test for called functions

File: src/components/homePage/Search/__tests__/Search.test.tsx [hunk 29]
```diff
@@ -0,0 +1,60 @@
...
+    fireEvent.change(input, { target: { value: "test" } });
+
+    //Assert
+    expect(tryToFindFilmsByTyipingMock).toHaveBeenCalledWith("test");
+    expect(tryToFindFilmsByTyipingMock).toBeCalledTimes(1);
+    expect(setSearchTextMock).toBeCalledTimes(1);
+    expect(setSearchTextMock).toHaveBeenCalledWith("test");
+  });
+
+  test("press button working", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/Search.test.tsx]: type: "Press button works"

File: src/components/homePage/Search/__tests__/FilmList.test.tsx [hunk 30]
```diff
@@ -0,0 +1,176 @@
...
+    fireEvent.click(navArrows[1]);
+
+    //Assert
+    expect(navArrows[0]).toBeInTheDocument();
+    expect(navArrows[1]).toBeInTheDocument();
+    expect(prevPageMock).toHaveBeenCalled();
+    expect(nextPageMock).toHaveBeenCalled();
+  });
+
+  test("pagination is not render", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/FilmList.test.tsx]: "Pagination not renders" or "Pagination is disabled"

File: src/components/homePage/Search/__tests__/FilmList.test.tsx [hunk 31]
```diff
@@ -0,0 +1,176 @@
...
+    {
+      Title: "Batman v Superman: Dawn of Justice (Ultimate Edition)",
+      Year: "2016",
+      imdbID: "tt18689424",
+      Type: "movie",
+      Poster:
+        "https://m.media-amazon.com/images/M/MV5BOTRlNWQwM2ItNjkyZC00MGI3LThkYjktZmE5N2FlMzcyNTIyXkEyXkFqcGdeQXVyMTEyNzgwMDUw._V1_SX300.jpg",
+    },
+  ];
+  test("component is render", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/FilmList.test.tsx]: "Component renders correctly"

File: src/components/homePage/Search/__tests__/FilmList.test.tsx [hunk 32]
```diff
@@ -0,0 +1,176 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import renderer from "react-test-renderer";
+import FilmList from "../FilmList";
+import { BrowserRouter } from "react-router-dom";
+
+describe("FilmList component", () => {
```
Comment (Anonim147) [inline src/components/homePage/Search/__tests__/FilmList.test.tsx]: FilmListC component tests

File: src/assets/icon/SearchArrow.jsx (hunk 1 above)
Comment (Yoland172) [reply src/assets/icon/SearchArrow.jsx]: fixed

File: src/assets/icon/SearchArrow.jsx (hunk 2 above)
Comment (Yoland172) [reply src/assets/icon/SearchArrow.jsx]: fixed

File: src/assets/icon/LoadingIcon.tsx (hunk 3 above)
Comment (Yoland172) [reply src/assets/icon/LoadingIcon.tsx]: fixed

File: src/assets/icon/leftArrow.jsx (hunk 4 above)
Comment (Yoland172) [reply src/assets/icon/leftArrow.jsx]: fixed

File: src/assets/icon/leftLongArrow.tsx (hunk 5 above)
Comment (Yoland172) [reply src/assets/icon/leftLongArrow.tsx]: fixed

File: src/components/filmPage/filmInfo/filmInfo.module.scss (hunk 6 above)
Comment (Yoland172) [reply src/components/filmPage/filmInfo/filmInfo.module.scss:112]: fixed

File: src/components/homePage/Search/__tests__/Search.test.tsx (hunk 7 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/Search.test.tsx]: fixed

File: src/components/homePage/Search/__tests__/Search.test.tsx (hunk 8 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/Search.test.tsx]: fixed

File: src/components/homePage/Search/__tests__/Search.test.tsx (hunk 9 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/Search.test.tsx]: fixed

File: src/components/homePage/Search/__tests__/Search.test.tsx (hunk 10 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/Search.test.tsx]: fixed

File: src/components/homePage/mainHeader/mainHeader.tsx (hunk 11 above)
Comment (Yoland172) [reply src/components/homePage/mainHeader/mainHeader.tsx]: fixed

File: src/components/homePage/mainHeader/mainHeader.tsx (hunk 12 above)
Comment (Yoland172) [reply src/components/homePage/mainHeader/mainHeader.tsx]: fixed

File: src/components/homePage/mainHeader/mainHeader.tsx (hunk 13 above)
Comment (Yoland172) [reply src/components/homePage/mainHeader/mainHeader.tsx]: fixed

File: src/assets/icon/SearchArrow.jsx (hunk 1 above)
Comment (Yoland172) [reply src/assets/icon/SearchArrow.jsx]: fixed

File: src/assets/icon/leftLongArrow.tsx (hunk 14 above)
Comment (Yoland172) [reply src/assets/icon/leftLongArrow.tsx]: fixed

File: src/assets/icon/rightArrow.jsx (hunk 15 above)
Comment (Yoland172) [reply src/assets/icon/rightArrow.jsx]: fixed

File: src/assets/icon/rightLongArrow.tsx (hunk 16 above)
Comment (Yoland172) [reply src/assets/icon/rightLongArrow.tsx]: fixed

File: src/assets/icon/socialMedia/InstagramIcon.tsx (hunk 17 above)
Comment (Yoland172) [reply src/assets/icon/socialMedia/InstagramIcon.tsx:2]: fixed

File: src/components/app/App.test.tsx (hunk 18 above)
Comment (Yoland172) [reply src/components/app/App.test.tsx]: fixed

File: src/components/filmPage/filmInfo/FilmInfo.tsx (hunk 19 above)
Comment (Yoland172) [reply src/components/filmPage/filmInfo/FilmInfo.tsx]: fixed

File: src/components/filmPage/filmInfo/FilmInfoContainer.tsx (hunk 20 above)
Comment (Yoland172) [reply src/components/filmPage/filmInfo/FilmInfoContainer.tsx]: fixed

File: src/components/filmPage/filmInfo/FilmInfoSkeleton.tsx (hunk 21 above)
Comment (Yoland172) [reply src/components/filmPage/filmInfo/FilmInfoSkeleton.tsx]: fixed

File: src/components/filmPage/filmInfo/filmInfo.module.scss (hunk 22 above)
Comment (Yoland172) [reply src/components/filmPage/filmInfo/filmInfo.module.scss]: fixed

File: src/components/homePage/Search/search.module.scss (hunk 23 above)
Comment (Yoland172) [reply src/components/homePage/Search/search.module.scss]: fixed

File: src/components/homePage/mainHeader/mainHeader.module.scss (hunk 24 above)
Comment (Yoland172) [reply src/components/homePage/mainHeader/mainHeader.module.scss]: fixed

File: src/index.tsx (hunk 26 above)
Comment (Yoland172) [reply src/index.tsx:10]: will not fix

File: src/components/filmPage/filmInfo/__tests__/filmInfo.test.tsx (hunk 27 above)
Comment (Yoland172) [reply src/components/filmPage/filmInfo/__tests__/filmInfo.test.tsx]: fixed

File: src/components/homePage/Search/__tests__/FilmList.test.tsx (hunk 28 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/FilmList.test.tsx]: fixed

File: src/components/homePage/Search/__tests__/Search.test.tsx (hunk 29 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/Search.test.tsx]: fixed

Comment (Yoland172) [reply src/components/homePage/Search/__tests__/Search.test.tsx]: fixed

File: src/components/homePage/Search/__tests__/FilmList.test.tsx (hunk 30 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/FilmList.test.tsx]: fixed

File: src/components/homePage/Search/__tests__/FilmList.test.tsx (hunk 31 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/FilmList.test.tsx]: fixed

File: src/components/homePage/Search/__tests__/FilmList.test.tsx (hunk 32 above)
Comment (Yoland172) [reply src/components/homePage/Search/__tests__/FilmList.test.tsx]: fixed

**Pull request merged**